pkg/didcomm/protocol/route/service.go

```diff
@@ -194,14 +216,22 @@ func (s *Service) handleRequest(msg *service.DIDCommMsg, myDID, theirDID string)
 
 func (s *Service) handleGrant(msg *service.DIDCommMsg) error {
 	// unmarshal the payload
-	grant := &Grant{}
+	grantMsg := &Grant{}
 
-	err := json.Unmarshal(msg.Payload, grant)
+	err := json.Unmarshal(msg.Payload, grantMsg)
 	if err != nil {
 		return fmt.Errorf("route grant message unmarshal : %w", err)
 	}
 
-	// TODO https://github.com/hyperledger/aries-framework-go/issues/948 integrate with framework components
+	// check if there are any channels registered for the message ID
+	grantCh := s.getRouteRegistrationCh(grantMsg.ID)
+
+	if grantCh != nil {
+		// invoke the channel for the incoming message
+		go func(grantCh chan Grant, grantMsg Grant) {
```

Comment:
It doesn't seem correct to spin a goroutine to send on a channel. Why isn't the channel buffered for this case?

Reply:
Agree. We don't need a go routine here as this logic is already running inside a goroutine. Updated the [code](https://github.com/hyperledger/aries-framework-go/compare/f2c71755a81cac382569f3cb14fae87828d66b2c..03621c455d43be8f44255077dc50427b5667fe6a#diff-7cc3df019152fe2c557826a312cc0a12R231).